enterprise/dev/ci/internal/ci/config.go

```diff
@@ -64,7 +64,12 @@ func NewConfig(now time.Time) Config {
 	var changedFiles []string
 	diffCommand := []string{"diff", "--name-only"}
 	if commit != "" {
-		diffCommand = append(diffCommand, "origin/main..."+commit)
+		if branch == "main" {
+			// On main, just look at the latest commit.
+			diffCommand = append(diffCommand, "@^")
```

Comment:
```suggestion
			// We run builds on every commit in main, so on main, just look at the diff of the current commit.
			diffCommand = append(diffCommand, "@^")
```

Another thought - maybe we could most the changed files generation after `runType` is evaluated, so we can use `runType.Is(MainBranch)` for the sake of consistency?